pkg/render/csi.go

```diff
@@ -261,14 +261,16 @@ func (c *csiComponent) csiTemplate() corev1.PodTemplateSpec {
 		Labels: templateLabels,
 	}
 	templateSpec := corev1.PodSpec{
-		Tolerations:      c.csiTolerations(),
-		Containers:       c.csiContainers(),
-		ImagePullSecrets: c.cfg.Installation.ImagePullSecrets,
-		Volumes:          c.csiVolumes(),
+		Tolerations:                  c.csiTolerations(),
+		Containers:                   c.csiContainers(),
+		ImagePullSecrets:             c.cfg.Installation.ImagePullSecrets,
+		Volumes:                      c.csiVolumes(),
+		AutomountServiceAccountToken: ptr.BoolToPtr(false),
 	}
 
 	if !c.cfg.Openshift && c.cfg.UsePSP {
 		templateSpec.ServiceAccountName = CSIDaemonSetName
+		templateSpec.AutomountServiceAccountToken = ptr.BoolToPtr(true)
```

Comment:
This change LGTM, but I think it could do with a code comment explaining that right now the only permission bound to the serviceaccount is the capability to use podsecuritypolicies, so we only need / want to mount in the serviceaccount token if PSP is enabled on the cluster.